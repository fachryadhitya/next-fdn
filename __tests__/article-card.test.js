import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import ArticleCard from '../components/article-card';

describe('Product Card Component', () => {
  it('Render successfully', () => {
    const { container } = render(<ArticleCard />);
    expect(container).toMatchSnapshot();
  });

  it('Render component with custom props', () => {
    const props = {
      image: 'https://static.femaledaily.com/dyn/640/images/prod-pics/product_1558000129_YOU_MAKEUP_800x800.png',
      author: 'Fachry',
      published_at: '2 hours ago',
      title: 'avoskin'
    };

    const { getByText, getByAltText } = render(<ArticleCard data={props} />);
    expect(getByText('Fachry')).toBeInTheDocument();
    expect(getByText('2 hours ago')).toBeInTheDocument();
    expect(getByText('avoskin')).toBeInTheDocument();
    const image = getByAltText('product picture');
    expect(image).toHaveAttribute('src', props.image);
  });
});
