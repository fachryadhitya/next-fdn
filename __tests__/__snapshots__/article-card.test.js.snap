// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card Component Render successfully 1`] = `
<div>
  <div
    class="my-4 w-full md:w-auto lg:w-32%"
  >
    <img
      alt="product picture"
      class="max-w-full h-auto rounded-lg object-center object-cover"
    />
    <div>
      <h3
        class="title-font font-bold capitalize text-xl"
      >
         
      </h3>
      <div
        class="flex gap-2"
      >
        <span
          class="text-black-200"
        >
           
        </span>
        <span
          class="text-gray-400"
        >
          |
        </span>
        <span
          class="text-neutral-400"
        />
      </div>
    </div>
  </div>
</div>
`;
